src/components/Language/Selector/MenuLanguageSelector.js

```diff
@@ -27,35 +28,52 @@ const applyAll = () => {
   return ALL_LANGUAGES;
 }
 
-const LanguageSelector = (selected, options, onLanguageChange, isAny, multiSelect, optionText, upward) => {
+const LanguageSelector = (selected, options, onLanguageChange, isAny, multiSelect, optionText, upward, isMobileDevice, uiDir) => {
   const onChange = (selected) => {
     onLanguageChange(selected);
   };
   const finalOptions = isAny ? [{text: 'Any', value: 'any'}] : options;
+  const value = multiSelect ? (isAny ? ['any'] : selected) : selected;
+
+  if (isMobileDevice && !multiSelect) {
+    return (
+      <select
+          className="language-mobile-select"
+          style={{direction: uiDir}}
+          value={value}
+          onChange={(event) => onChange(event.target.value)}>
+          { options.map(x => <option key={`opt-${x.value}`} value={x.value}>{x.name}</option>) }
+      </select>
+    );
+  }
+
   return (
     <Dropdown
       selection
       upward={upward}
       multiple={multiSelect}
-      value={multiSelect ? (isAny ? ['any'] : selected) : selected}
+      value={value}
       onChange={(event, data) => onChange(data.value)}
       options={options}
     />
   );
 }
 
 const MenuLanguageSelector = ({ languages = [], selected = [], onLanguageChange = noop, multiSelect = true, optionText = null, upward = false}) => {
+  const uiDir = useSelector(state => settings.getUIDir(state.settings));
+  const { isMobileDevice } = useContext(DeviceInfoContext);
   const contentLanguages = useSelector(state => settings.getContentLanguages(state.settings));
   const onLanguageChangeInternal = selected => {
     onLanguageChange(selected);
   };
 
   const validLanguages = languages.filter(lang => contentLanguages.includes(lang));
   const otherLanguages = languages.filter(lang => !contentLanguages.includes(lang));
-  const options = getOptions({ languages: validLanguages }).concat([{value: 'divider', className: 'language-selection-divider disabled'}]).concat(getOptions({ languages: otherLanguages}));
+  const dividerArray = !isMobileDevice || multiSelect ? [{value: 'divider', className: 'language-selection-divider disabled'}] : [];
+  const options = getOptions({ languages: validLanguages }).concat(dividerArray).concat(getOptions({ languages: otherLanguages }));
   // Special case when all laguages are selected, e.g., show content with any language.
   const isAny = languages === selected;
-  return LanguageSelector(selected, options, onLanguageChangeInternal, isAny, multiSelect, optionText, upward);
+  return LanguageSelector(selected, options, onLanguageChangeInternal, isAny, multiSelect, optionText, upward, isMobileDevice, uiDir);
```

Comment:
are you sure to use as function here?
https://dev.to/igor_bykov/react-calling-functional-components-as-functions-1d3l#:~:text=gets%20re%2Dmounted.-,Conclusion,-A%20function%20that

Reply:
Fixed.